code/modules/reagents/chemistry/reagents/alchemical_potions.dm

```diff
@@ -0,0 +1,242 @@
+///////////////////////////////////////////////////////////////////
+					//Noita Potions
+//////////////////////////////////////////////////////////////////
+
+#define MAGIC_REAGENT_TOUCH 4
+#define TELEPORTARIUM_RANGE 4
+#define TELEPORTARIUM_CYCLE 2
+#define POLYMORPHIUM_DURATION 15 SECONDS
+
+/datum/reagent/magic
+	name = "alchemic precursor"
+	description = "Basic component in all alchemical potions."
+	random_unrestricted = FALSE
+	metabolization_rate = 3 * REAGENTS_METABOLISM
+	color = "#1C2EC3"
+
+//	----	POLYMORPHINE	----
+
+/datum/reagent/magic/polymorphine
+	name = "polymorphine"
+	description = "Magic potion that transforms you into a harmless animal."
+	color = "#9D5A99"
+	taste_description = "a rainbow of tastes"
+	overdose_threshold = 25
+	metabolization_rate = 0.1 * REAGENTS_METABOLISM
+	var/list_mobs = FRIENDLY_SPAWN
+	var/obj/shapeshift_holder/shapeshiftdata
+
+/datum/reagent/magic/polymorphine/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if (reac_volume >= overdose_threshold && (method == TOUCH || method == VAPOR))
+		polymorph_target(M)
+	..()
+
+/datum/reagent/magic/polymorphine/overdose_start(mob/living/L)
+	..()
+	polymorph_target(L)
+	metabolization_rate = 10 * REAGENTS_METABOLISM
+
+/datum/reagent/magic/polymorphine/proc/polymorph_target(mob/living/L)
+	shapeshiftdata = locate() in L
+	if(shapeshiftdata)
+		return
+	var/mob/living/shape = create_random_mob((get_turf(L)), list_mobs)
+	shapeshiftdata = new(shape,null,L)
+	addtimer(CALLBACK(shapeshiftdata, /obj/shapeshift_holder.proc/restore), POLYMORPHIUM_DURATION)
+
+/datum/reagent/magic/polymorphine/on_mob_end_metabolize(mob/living/L)
+	..()
+	if(!shapeshiftdata)
+		return
+	shapeshiftdata.restore()
+
+/datum/reagent/magic/polymorphine/chaotic
+	name = "chaotic polymorphine"
+	description = "Magic potion that transforms you into a hostile creature."
+	list_mobs = HOSTILE_SPAWN
+
+/datum/reagent/magic/polymorphine/unstable
+	name = "unstable polymorphine"
+	description = "Magic potion that transforms you into a random creature."
+
+/datum/reagent/magic/polymorphine/unstable/reaction_mob(mob/living/M, method=TOUCH, reac_volume)	
+	list_mobs = prob(50) ? HOSTILE_SPAWN : FRIENDLY_SPAWN
+	..()
+
+/datum/reagent/magic/polymorphine/unstable/on_mob_metabolize(mob/living/L)
+	..()
+	list_mobs = prob(50) ? HOSTILE_SPAWN : FRIENDLY_SPAWN
+
+//	----	BERSERKIUM	----
+
+/datum/reagent/magic/berserkium
+	name = "berserkium"
+	description = "Magic potion that enrages you when drunk."
+	color = "#A14444"
+	taste_description = "something that makes you angry"
```

Comment:
Guess it's removed now anyway :(

Reply:
I removed them because I didn't like how the ones I added sounded.

But I will add them back if you can help me come up with good tastes for everything.